openmeter/ent/schema/subscription.go

```diff
@@ -0,0 +1,141 @@
+package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/dialect"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+
+	"github.com/openmeterio/openmeter/openmeter/subscription"
+	"github.com/openmeterio/openmeter/pkg/currencyx"
+	"github.com/openmeterio/openmeter/pkg/datex"
+	"github.com/openmeterio/openmeter/pkg/framework/entutils"
+)
+
+type Subscription struct {
+	ent.Schema
+}
+
+func (Subscription) Mixin() []ent.Mixin {
+	return []ent.Mixin{
+		entutils.IDMixin{},
+		entutils.NamespaceMixin{},
+		entutils.TimeMixin{},
+		entutils.MetadataAnnotationsMixin{},
+		entutils.CadencedMixin{},
+	}
+}
+
+func (Subscription) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("plan_key").NotEmpty().Immutable(),
+		field.Int("plan_version").Min(1).Immutable(),
```

Comment:
Are we using `plan_key` + `plan_version` instead of `plan_id` to know for version migration which plan it is?

Reply:
Just that it conveys more visible meaning than an opaque id.